contract_comm/caller.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 The Celo Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package contract_comm
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/celo-org/celo-blockchain/accounts/abi"
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/common/hexutil"
+	ccerrors "github.com/celo-org/celo-blockchain/contract_comm/errors"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+	"github.com/celo-org/celo-blockchain/log"
+	"github.com/celo-org/celo-blockchain/metrics"
+)
+
+func GetRegisteredAddress(registryId common.Hash, header *types.Header, state vm.StateDB) (*common.Address, error) {
+	cc, err := NewCaller(header, state)
+	if err != nil {
+		return &common.Address{}, err
+	}
+	return cc.GetRegisteredAddress(registryId)
+}
+
+// Contract Communicator Implementation
+func (c contractCommunicator) MakeStaticCallWithAddress(contractAddress common.Address, abi abi.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64) (uint64, error) {
+	evm, err := c.builder.createEVM()
+	if err != nil {
+		return 0, err
+	}
+	return evm.StaticCallFromSystem(contractAddress, abi, funcName, args, returnObj, gas)
+}
+
+func (c contractCommunicator) MakeCallWithAddress(contractAddress common.Address, abi abi.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64, value *big.Int) (uint64, error) {
```

Comment:
My initial thought in reading `MakeCallWithAddress` was that the call would be made _from_ the given address rather than to it. You might consider renaming this to `MakeCallToAddress`